src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,168 @@
+package duke.task;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+import duke.command.DukeException;
+
+/**
+ * This class implements a TaskList object that contains the task list and its operations.
+ *
+ * CS2103T ip
+ * AY21/22 Semester 1
+ *
+ * @author Kishendran Vendar Kon (Group G05)
+ */
+public class TaskList {
+    /** Formatter to change String to a Date */
+    protected static SimpleDateFormat formatter = new SimpleDateFormat("MMM dd yyyy hh:mm aaa");
+    private List<Task> toDoList;
+
+    /**
+     * Default constructor.
+     *
+     * @param tasks List of tasks to be loaded.
+     */
+    public TaskList(ArrayList<String> tasks) {
+        this.toDoList = new ArrayList<>();
+        for (int i = 0; i < tasks.size(); i++) {
+            String s = tasks.get(i);
```

Comment:
Maybe you can use enhanced `for` instead of the normal one?
```suggestion
        for (String s : tasks) {
```